lucet-module/src/functions.rs

```diff
@@ -125,9 +125,15 @@ impl OwnedFunctionMetadata {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
 pub struct FunctionHandle {
     pub ptr: FunctionPointer,
     pub id: FunctionIndex,
+    /// This is `true` if and only if this handle was referenced through the start section.
+    ///
+    /// The same function may be referenced via other contexts, and will appear with `false` in this
+    /// field in those cases.
+    pub is_start_func: bool,
```

Comment:
This is a little weird; I could be convinced this belongs in `lucet-runtime-internals` only, but since `FunctionHandle` is only used downstream of `lucet-module` it seemed okay.